### ISSUE TITLE
Expose resources for awx-controller-manager in chart

### ISSUE THREAD

**Issue by labmonkey42** [ansible-community/awx-operator-helm#14]
Migrating from https://github.com/ansible/awx-operator/issues/1919

Summary

As noted in https://github.com/ansible/awx-operator/issues/1115, GKE Autopilot automatically changes pod resource requests and/or limits. The specific defect described in that issue occurs because Autopilot clusters with the `bursting` feature `disabled` automatically (and transparently) change pod `resources` to set `limits` equal to the `requests` value when both values are defined. This means that `awx-operator-controller-manager` is left with insufficient memory, and will always be `OOMKilled`.

In my case, I deploy the `awx-operator` helm chart with an Ansible playbook, and thus I can use a `kubernetes.core.k8s_json_patch` task on the `Deployment` to set the `requests` value equal to the `limits` value immediately after the helm chart is deployed, effectively working around the problem.  However, this does introduce the issue that any change that redeploys the chart will also have to re-patch the Deployment.

What https://github.com/ansible/awx-operator/issues/1115 didn't spell out, and I'm specifically requesting here, is the ability to adjust the `resources` `requests` and `limits` for the `awx-manager` pod in the `Deployment` from the helm chart values directly. This would allow us to handle this issue before deploying the chart, and prevent the play from repeatedly adjusting these values in subsequent runs.

Basically, as D1StrX notes there, the CRD AWX should include a section for `awx_manager_resource_requirements`.

**Comment by oraNod** [comment]
Thanks for migrating the issue to this repo @labmonkey42

**Comment by miles-w-3** [comment]
I believe it is possible to do this by writing the deployment code to a tpl file, and then parsing it before it lands on the cluster. This will allow the use of helm values to modify the kustomize output for the controller deployment, as opposed to how it is currently used now, which is simply just to create values

**Comment by ChrisLee-JackHenry** [comment]
Any ETA on a release with these changes?

**Comment by oraNod** [comment]
> Any ETA on a release with these changes?

I'd like to get a release cut next week. No promises though. We can use this issue to figure out the process and track the effort: https://github.com/ansible-community/awx-operator-helm/issues/30

**Comment by ChrisLee-JackHenry** [comment]
> > Any ETA on a release with these changes?
> 
> I'd like to get a release cut next week. No promises though. We can use this issue to figure out the process and track the effort: [#30](https://github.com/ansible-community/awx-operator-helm/issues/30)

The changes merged in #30 don't bump the chart version, so it wasn't released.
